命令行识别特长音频（实测2小时以上）导致接口错误

用命令行识别特长音频（实测2小时以上）的时候，会出现后续查询状态过于频繁导致接口错误。
这个是否可以设置一个参数？或者应该在哪里修改呢？

2024-03-15 13:55:50,806 - [INFO] 加载文件成功: 1710482150.aac
2024-03-15 13:55:50,960 - [INFO] 申请上传成功, 总计大小117881KB, 24分片, 分片大小5120KB: biz376761249775503627/AIA504914598776816296.AAC
2024-03-15 13:55:50,960 - [INFO] 开始上传分片0: 0-5242880
2024-03-15 13:55:51,932 - [INFO] 分片0上传成功: 3f2e475b1f9af3e09341282eeab97f5e
2024-03-15 13:55:51,933 - [INFO] 开始上传分片1: 5242880-10485760
2024-03-15 13:55:52,721 - [INFO] 分片1上传成功: 7b18cd42af41414b6fdd163b87d6b33f
2024-03-15 13:55:52,721 - [INFO] 开始上传分片2: 10485760-15728640
2024-03-15 13:55:53,143 - [INFO] 分片2上传成功: bb5609e07bd29ab7358e0b4ec237db2c
2024-03-15 13:55:53,144 - [INFO] 开始上传分片3: 15728640-20971520
2024-03-15 13:55:53,604 - [INFO] 分片3上传成功: 4ce89bafb8717047610abe60b3b2a792
2024-03-15 13:55:53,604 - [INFO] 开始上传分片4: 20971520-26214400
2024-03-15 13:55:53,942 - [INFO] 分片4上传成功: 757502ae6b330b457cee724112f62496
2024-03-15 13:55:53,943 - [INFO] 开始上传分片5: 26214400-31457280
2024-03-15 13:55:54,277 - [INFO] 分片5上传成功: c3aab1007f3a3739c6cb727f5fbd1d6d
2024-03-15 13:55:54,278 - [INFO] 开始上传分片6: 31457280-36700160
2024-03-15 13:55:54,640 - [INFO] 分片6上传成功: aeae88d3b37548fb123d9fe066137f69
2024-03-15 13:55:54,641 - [INFO] 开始上传分片7: 36700160-41943040
2024-03-15 13:55:54,910 - [INFO] 分片7上传成功: 327d8ba25d26e00f21231cbd9d2b6636
2024-03-15 13:55:54,910 - [INFO] 开始上传分片8: 41943040-47185920
2024-03-15 13:55:55,289 - [INFO] 分片8上传成功: 62a35c23d874677622bb685cbeb10662
2024-03-15 13:55:55,290 - [INFO] 开始上传分片9: 47185920-52428800
2024-03-15 13:55:55,536 - [INFO] 分片9上传成功: 599185d1f8ed85399c923a59ac0217c5
2024-03-15 13:55:55,536 - [INFO] 开始上传分片10: 52428800-57671680
2024-03-15 13:55:55,878 - [INFO] 分片10上传成功: fe0c2087258f027348e481074ad3d286
2024-03-15 13:55:55,878 - [INFO] 开始上传分片11: 57671680-62914560
2024-03-15 13:55:56,133 - [INFO] 分片11上传成功: 8c41c3a2f621188cc6d998a802710186
2024-03-15 13:55:56,133 - [INFO] 开始上传分片12: 62914560-68157440
2024-03-15 13:55:56,466 - [INFO] 分片12上传成功: 3b4ffcca16fc9ef53ed00a9ff76980bd
2024-03-15 13:55:56,466 - [INFO] 开始上传分片13: 68157440-73400320
2024-03-15 13:55:56,729 - [INFO] 分片13上传成功: 661b16a4d76dae8fb3ec8224dab4b25f
2024-03-15 13:55:56,729 - [INFO] 开始上传分片14: 73400320-78643200
2024-03-15 13:55:57,053 - [INFO] 分片14上传成功: 3b917d218fcd366daff82fcd0d0815a2
2024-03-15 13:55:57,053 - [INFO] 开始上传分片15: 78643200-83886080
2024-03-15 13:55:57,310 - [INFO] 分片15上传成功: 23555840c3a3cff47fc3e4c279f08cc3
2024-03-15 13:55:57,311 - [INFO] 开始上传分片16: 83886080-89128960
2024-03-15 13:55:57,803 - [INFO] 分片16上传成功: e8ba709faa228c26b02d248dbbbe3e4a
2024-03-15 13:55:57,804 - [INFO] 开始上传分片17: 89128960-94371840
2024-03-15 13:55:58,679 - [INFO] 分片17上传成功: f3485f4ed8c9c3aa243215ef171590d9
2024-03-15 13:55:58,679 - [INFO] 开始上传分片18: 94371840-99614720
2024-03-15 13:55:59,201 - [INFO] 分片18上传成功: 184cf4da6b46520ea77a5e6d9f8c1963
2024-03-15 13:55:59,201 - [INFO] 开始上传分片19: 99614720-104857600
2024-03-15 13:55:59,580 - [INFO] 分片19上传成功: 824b6ab5ff2bcd20db18500648e1eed4
2024-03-15 13:55:59,581 - [INFO] 开始上传分片20: 104857600-110100480
2024-03-15 13:55:59,946 - [INFO] 分片20上传成功: 234139282ea0515dce14e97a4b815af4
2024-03-15 13:55:59,946 - [INFO] 开始上传分片21: 110100480-115343360
2024-03-15 13:56:00,251 - [INFO] 分片21上传成功: c20b267d5d1e34f3a049b51e9e95f00a
2024-03-15 13:56:00,251 - [INFO] 开始上传分片22: 115343360-120586240
2024-03-15 13:56:00,557 - [INFO] 分片22上传成功: 3f43fe9966cfc70b9c21ea0a5e967e53
2024-03-15 13:56:00,558 - [INFO] 开始上传分片23: 120586240-125829120
2024-03-15 13:56:00,616 - [INFO] 分片23上传成功: 26ef1baa159576e8384cddfbb03dfb72
2024-03-15 13:56:00,712 - [INFO] 提交成功
2024-03-15 13:56:00,764 - [INFO] 任务已创建: asr504914615134632045
2024-03-15 13:56:00,819 - [INFO] 等待识别开始
2024-03-15 13:56:01,884 - [INFO] 等待识别开始
2024-03-15 13:56:02,940 - [INFO] 等待识别开始
2024-03-15 13:56:04,004 - [INFO] 等待识别开始
2024-03-15 13:56:05,062 - [INFO] 等待识别开始
2024-03-15 13:56:06,127 - [INFO] 等待识别开始
2024-03-15 13:56:07,184 - [INFO] 等待识别开始
2024-03-15 13:56:08,241 - [INFO] 等待识别开始
2024-03-15 13:56:09,300 - [INFO] 等待识别开始
2024-03-15 13:56:10,366 - [INFO] 等待识别开始
2024-03-15 13:56:11,426 - [INFO] 等待识别开始
2024-03-15 13:56:12,491 - [INFO] 等待识别开始
2024-03-15 13:56:13,557 - [INFO] 等待识别开始
2024-03-15 13:56:14,615 - [INFO] 等待识别开始
2024-03-15 13:56:15,669 - [INFO] 等待识别开始
2024-03-15 13:56:16,731 - [INFO] 等待识别开始
2024-03-15 13:56:17,798 - [INFO] 等待识别开始
2024-03-15 13:56:18,861 - [INFO] 等待识别开始
2024-03-15 13:56:19,914 - [INFO] 等待识别开始
2024-03-15 13:56:20,967 - [INFO] 等待识别开始
2024-03-15 13:56:22,024 - [INFO] 等待识别开始
2024-03-15 13:56:23,084 - [INFO] 等待识别开始
2024-03-15 13:56:24,135 - [INFO] 等待识别开始
2024-03-15 13:56:25,193 - [INFO] 等待识别开始
2024-03-15 13:56:26,266 - [INFO] 等待识别开始
2024-03-15 13:56:27,342 - [INFO] 等待识别开始
2024-03-15 13:56:28,403 - [INFO] 等待识别开始
2024-03-15 13:56:29,461 - [INFO] 等待识别开始
2024-03-15 13:56:30,522 - [INFO] 等待识别开始
2024-03-15 13:56:31,581 - [INFO] 等待识别开始
2024-03-15 13:56:32,649 - [INFO] 等待识别开始
2024-03-15 13:56:33,703 - [INFO] 等待识别开始
2024-03-15 13:56:34,759 - [INFO] 识别中-RUNING
2024-03-15 13:56:35,822 - [INFO] 识别中-RUNING
2024-03-15 13:56:36,875 - [INFO] 识别中-RUNING
2024-03-15 13:56:37,937 - [INFO] 识别中-RUNING
2024-03-15 13:56:38,993 - [INFO] 识别中-RUNING
2024-03-15 13:56:40,058 - [INFO] 识别中-RUNING
2024-03-15 13:56:41,120 - [INFO] 识别中-RUNING
2024-03-15 13:56:42,188 - [INFO] 识别中-RUNING
2024-03-15 13:56:43,243 - [INFO] 识别中-RUNING
2024-03-15 13:56:44,296 - [INFO] 识别中-RUNING
2024-03-15 13:56:45,354 - [INFO] 识别中-RUNING
2024-03-15 13:56:46,416 - [INFO] 识别中-RUNING
2024-03-15 13:56:47,477 - [INFO] 识别中-RUNING
2024-03-15 13:56:48,540 - [INFO] 识别中-RUNING
2024-03-15 13:56:49,595 - [INFO] 识别中-RUNING
2024-03-15 13:56:50,657 - [INFO] 识别中-RUNING
2024-03-15 13:56:51,725 - [INFO] 识别中-RUNING
2024-03-15 13:56:52,791 - [INFO] 识别中-RUNING
2024-03-15 13:56:53,848 - [INFO] 识别中-RUNING
2024-03-15 13:56:54,901 - [INFO] 识别中-RUNING
2024-03-15 13:56:55,959 - [INFO] 识别中-RUNING
2024-03-15 13:56:57,019 - [INFO] 识别中-RUNING
2024-03-15 13:56:58,091 - [INFO] 识别中-RUNING
2024-03-15 13:56:59,146 - [INFO] 识别中-RUNING
2024-03-15 13:57:00,197 - [INFO] 识别中-RUNING
2024-03-15 13:57:01,255 - [INFO] 识别中-RUNING
2024-03-15 13:57:02,311 - [INFO] 识别中-RUNING
2024-03-15 13:57:03,371 - [INFO] 识别中-RUNING
2024-03-15 13:57:04,429 - [INFO] 识别中-RUNING
2024-03-15 13:57:05,484 - [INFO] 识别中-RUNING
2024-03-15 13:57:06,561 - [INFO] 识别中-RUNING
2024-03-15 13:57:07,617 - [INFO] 识别中-RUNING
2024-03-15 13:57:08,679 - [INFO] 识别中-RUNING
2024-03-15 13:57:09,740 - [INFO] 识别中-RUNING
2024-03-15 13:57:10,798 - [INFO] 识别中-RUNING
2024-03-15 13:57:11,862 - [INFO] 识别中-RUNING
2024-03-15 13:57:12,914 - [INFO] 识别中-RUNING
2024-03-15 13:57:13,969 - [INFO] 识别中-RUNING
2024-03-15 13:57:15,028 - [INFO] 识别中-RUNING
2024-03-15 13:57:16,089 - [INFO] 识别中-RUNING
2024-03-15 13:57:17,149 - [INFO] 识别中-RUNING
2024-03-15 13:57:18,201 - [INFO] 识别中-RUNING
2024-03-15 13:57:19,253 - [INFO] 识别中-RUNING
2024-03-15 13:57:20,312 - [INFO] 识别中-RUNING
2024-03-15 13:57:21,505 - [ERROR] 接口错误: -509:请求过于频繁，请稍后再试

--- a/README.md
+++ b/README.md
@@ -40,9 +40,15 @@ bcut_asr video.mp4 subtitle.srt
 bcut_asr video.mp4 -f srt - > subtitle.srt
 ```
 
+长音频指定任务状态轮询间隔(秒)，避免接口频繁调用
+
+```bash
+bcut_asr video.mp4 -f srt -i 30 - > subtitle.srt
 ```
-bcut_asr -h                
-usage: bcut_asr [-h] [-f [{srt,json,lrc,txt}]] input [output]
+
+```
+bcut_asr -h
+usage: bcut-asr [-h] [-f [{srt,json,lrc,txt}]] [-i [1.0]] input [output]
 
 必剪语音识别
 
@@ -54,6 +60,8 @@ options:
   -h, --help            show this help message and exit
   -f [{srt,json,lrc,txt}], --format [{srt,json,lrc,txt}]
                         输出字幕格式
+  -i [1.0], --interval [1.0]
+                        任务状态轮询间隔(秒)
 
 支持输入音频格式: flac, aac, m4a, mp3, wav 支持自动调用ffmpeg提取视频伴音
 ```
@@ -91,7 +99,7 @@ from argparse import Namespace
 
 
 f = open("file.mp4", "rb")
-argg = Namespace(format="srt", input=f, output=None)
+argg = Namespace(format="srt", interval=30.0, input=f, output=None)
 run_everywhere(argg)
 
 ```

--- a/bcut_asr/__init__.py
+++ b/bcut_asr/__init__.py
@@ -81,6 +81,9 @@ def run_everywhere(argg):
                 logging.error('输出格式错误')
                 sys.exit(-1)
 
+    interval = argg.interval
+    if interval is None:
+        interval = 30.0
     # 开始执行转换逻辑
     asr = BcutASR()
     asr.set_data(raw_data=infile_data, data_fmt=infile_fmt)
@@ -107,7 +110,7 @@ def run_everywhere(argg):
                     # 识别成功, 回读字幕数据
                     result = task_resp.parse()
                     break
-            time.sleep(300.0)
+            time.sleep(interval)
         if not result.has_data():
             logging.error('未识别到语音')
             sys.exit(-1)
@@ -120,6 +123,7 @@ def run_everywhere(argg):
                 outfile.write(result.json())
             case 'txt':
                 outfile.write(result.to_txt())
+        outfile.close()
         logging.info(f'转换成功: {outfile_name}')
     except APIError as err:
         logging.error(f'接口错误: {err.__str__()}')
@@ -149,16 +153,16 @@ class BcutASR:
     __etags: list[str]
     __download_url: str
     task_id: str
-    
+
     def __init__(self, file: Optional[str | PathLike] = None) -> None:
         self.session = requests.Session()
         self.task_id = None
         self.__etags = []
         if file:
             self.set_data(file)
-    
+
     def set_data(self,
-        file: Optional[str | PathLike] = None, 
+        file: Optional[str | PathLike] = None,
         raw_data: Optional[bytes] = None,
         data_fmt: Optional[SUPPORT_SOUND_FORMAT] = None
     ) -> None:
@@ -182,7 +186,7 @@ class BcutASR:
             raise TypeError('format is not support')
         self.sound_fmt = suffix
         logging.info(f'加载文件成功: {self.sound_name}')
-    
+
     def upload(self) -> None:
         '申请上传'
         if not self.sound_bin or not self.sound_fmt:
@@ -209,7 +213,7 @@ class BcutASR:
         logging.info(f'申请上传成功, 总计大小{resp_data.size // 1024}KB, {self.__clips}分片, 分片大小{resp_data.per_size // 1024}KB: {self.__in_boss_key}')
         self.__upload_part()
         self.__commit_upload()
-        
+
     def __upload_part(self) -> None:
         '上传音频数据'
         for clip in range(self.__clips):
@@ -223,7 +227,7 @@ class BcutASR:
             etag = resp.headers.get('Etag')
             self.__etags.append(etag)
             logging.info(f'分片{clip}上传成功: {etag}')
-    
+
     def __commit_upload(self) -> None:
         '提交上传数据'
         resp = self.session.post(API_COMMIT_UPLOAD, data={
@@ -241,7 +245,7 @@ class BcutASR:
         resp_data = ResourceCompleteRspSchema.parse_obj(resp['data'])
         self.__download_url = resp_data.download_url
         logging.info(f'提交成功')
-    
+
     def create_task(self) -> str:
         '开始创建转换任务'
         resp = self.session.post(API_CREATE_TASK, json={
@@ -257,7 +261,7 @@ class BcutASR:
         self.task_id = resp_data.task_id
         logging.info(f'任务已创建: {self.task_id}')
         return self.task_id
-    
+
     def result(self, task_id: Optional[str] = None) -> ResultRspSchema:
         '查询转换结果'
         resp = self.session.get(API_QUERY_RESULT, params={

--- a/bcut_asr/__main__.py
+++ b/bcut_asr/__main__.py
@@ -16,6 +16,7 @@ parser = ArgumentParser(
     epilog=f"支持输入音频格式: {', '.join(INFILE_FMT)}  支持自动调用ffmpeg提取视频伴音"
 )
 parser.add_argument('-f', '--format', nargs='?', default='srt', choices=OUTFILE_FMT, help='输出字幕格式')
+parser.add_argument('-i', '--interval', nargs='?', type=float, default='1.0', metavar='1.0', help='任务状态轮询间隔(秒)')
 parser.add_argument('input', type=FileType('rb'), help='输入媒体文件')
 parser.add_argument('output', nargs='?', type=FileType('w', encoding='utf8'), help='输出字幕文件, 可stdout')
 
@@ -80,6 +81,9 @@ def main():
                 logging.error('输出格式错误')
                 sys.exit(-1)
 
+    interval = args.interval
+    if interval is None:
+        interval = 1.0
     # 开始执行转换逻辑
     asr = BcutASR()
     asr.set_data(raw_data=infile_data, data_fmt=infile_fmt)
@@ -106,7 +110,7 @@ def main():
                     # 识别成功, 回读字幕数据
                     result = task_resp.parse()
                     break
-            time.sleep(1.0)
+            time.sleep(interval)
         if not result.has_data():
             logging.error('未识别到语音')
             sys.exit(-1)
@@ -119,6 +123,7 @@ def main():
                 outfile.write(result.json())
             case 'txt':
                 outfile.write(result.to_txt())
+        outfile.close()
         logging.info(f'转换成功: {outfile_name}')
     except APIError as err:
         logging.error(f'接口错误: {err.__str__()}')

